File: src/main/java/io/chesslave/hands/DragBotMover.java
package io.chesslave.hands;

import io.chesslave.model.Square;
import io.chesslave.visual.BoardImage;

/**
 * A bot able to move pieces through a drag and drop strategy.
 */
public class DragBotMover extends BaseBotMover {

    public DragBotMover(BoardImage board) {
        super(board);
    }

    @Override
    public void move(Square from, Square to) throws MoverException {
        try {
            pointer.dragFrom(getSquareCoords(from));
            pointer.dropAt(getSquareCoords(to));
        } catch (RuntimeException re) {
            throw new MoverException(re);
        }
    }
}

File: backend/src/main/java/io/chesslave/hands/DragBotMover.kt
package io.chesslave.hands

import io.chesslave.model.Square
import io.chesslave.visual.model.BoardImage

/**
 * A bot able to move pieces through a drag and drop strategy.
 */
class DragBotMover(val pointer: Pointer, board: BoardImage) : BaseBotMover(board) {

    override fun move(from: Square, to: Square) {
        try {
            pointer.dragFrom(getSquareCoords(from))
            pointer.dropAt(getSquareCoords(to))
        } catch (re: RuntimeException) {
            throw MoverException(re)
        }
    }
}
